src/resources/GenerateNewPlan.tsx

```diff
@@ -63,9 +63,9 @@ const GenerateNewPlan: FC = () => {
           searchYear: newPlan.years[0]._id,
         }),
       );
-      // Make a new distribution for each distribution of the major of the plan.
-      toAddMajor.distributions.forEach(
```

Comment:
I think forEach loop does not support async/await

Reply:
This will cause distributions to be added to redux before all of them are actually fetched